src/editor/Gui.cpp

```diff
@@ -496,6 +496,116 @@ void Gui::drawMenuBar() {
 			//map->write("D:/Steam/steamapps/common/Sven Co-op/svencoop_addon/maps/yabma_move.bsp");
 			map->write(map->path);
 		}
+		if (ImGui::BeginMenu("Export")) {
+			if (ImGui::MenuItem("Entity file", NULL)) {
+				Bsp* map = app->getMapContainingCamera()->map;
+				if (map)
+				{
+					logf("Export entities: %s%s%s\n", g_settings.gamedir.c_str(), g_settings.workingdir.c_str(), "entities.ent");
+					ofstream entFile(g_settings.gamedir + g_settings.workingdir + "entities.ent", ios::out | ios::trunc);
+					entFile.write((const char*)map->lumps[LUMP_ENTITIES], map->header.lump[LUMP_ENTITIES].nLength);
+				}
+			}
+			if (ImGui::MenuItem("Wad file", NULL)) {
+				Bsp* map = app->getMapContainingCamera()->map;
+				if (map)
+				{
+					logf("Export wad: %s%s%s\n", g_settings.gamedir.c_str(), g_settings.workingdir.c_str(), (map->name + ".wad").c_str());
+					if (map->textureCount > 0)
+					{
+						Wad* tmpWad = new Wad(map->path);
+						std::vector<WADTEX*> tmpWadTex;
+						for (int i = 0; i < map->textureCount; i++) {
+							int32_t oldOffset = ((int32_t*)map->textures)[i + 1];
+							BSPMIPTEX* bspTex = (BSPMIPTEX*)(map->textures + oldOffset);
+							WADTEX* oldTex = new WADTEX(bspTex);
+							tmpWadTex.push_back(oldTex);
+						}
+						tmpWad->write(g_settings.gamedir.c_str() + (g_settings.workingdir.c_str() + map->name) + ".wad", &tmpWadTex[0], tmpWadTex.size());
+						for (int i = 0; i < map->textureCount; i++) {
+							delete tmpWadTex[i];
+						}
+						tmpWadTex.clear();
+					}
+					else
+					{
+						logf("No textures for export.\n");
+					}
+				}
+			}
+			ImGui::EndMenu();
+		}
+		if (ImGui::BeginMenu("Import")) {
+			if (ImGui::MenuItem("Entity file", NULL)) {
+				Bsp* map = app->getMapContainingCamera()->map;
+				if (map)
+				{
+					logf("Import entities from: %s%s%s\n", g_settings.gamedir.c_str(), g_settings.workingdir.c_str(), "entities.ent");
+					if (fileExists(g_settings.gamedir + g_settings.workingdir + "entities.ent"))
+					{
+						int datasize = fileSize(g_settings.gamedir + g_settings.workingdir + "entities.ent");
+						char* data = new char[datasize + 1];
+						data[datasize] = '\0';
+						ifstream entFile(g_settings.gamedir + g_settings.workingdir + "entities.ent", std::ios::binary);
+						entFile.read(data, datasize);
+						map->replace_lump(LUMP_ENTITIES, data, datasize);
+					}
+					else
+					{
+						logf("Error! No file!\n");
+					}
+				}
+			}
+			if (ImGui::MenuItem("Wad file", NULL)) {

```

Comment:
This still seems not to work with the latest code. I exported the WAD from this map, which has some embedded textures:
[duhhh.zip](https://github.com/wootguy/bspguy/files/6040454/duhhh.zip)
Then I exported the WAD, and edited the `duh_rock1` texture in wally, and saved the WAD.
![image](https://user-images.githubusercontent.com/12087544/109102582-0b56fe80-76de-11eb-87be-fe57dbc34905.png)
After importing the WAD with bspguy, the rock textures doesn't change.

Reply:
@wootguy yes because internal textures with more priority or duplicated ? Possible need rename 'import' to 'merge with' and adding some fixes in code :)

Reply:
Oh ok, so importing doesn't replace textures, it appends to the existing set of textures.

I was thinking it would replace any textures that have the same name. It's ok to leave it the way it is now, but please add a tooltip explaining what it does.

Reply:
Fixed. Now it reeplace textures and append if no found.